S3Common.cc

```diff
@@ -0,0 +1,331 @@
+#include <atomic>
+#include <cassert>
+#include <chrono>
+#include <fstream>
+#include <functional>
+#include <iostream>
+#include <thread>
+#include <variant>
+#include <random>
+
+#include "libs3.h"
+#include "tbb/concurrent_queue.h"
+#include "S3Common.h"
+
+
+namespace cce::tf {
+
+class S3LibWrapper {
+  public:
+    S3LibWrapper(bool async=false) : async_(async), running_(false) {
+      initStatus_ = S3_initialize("s3", S3_INIT_ALL, "");
+      if ( initStatus_ != S3StatusOK ) {
+        std::cerr << "Failed to initialize libs3, error: " << S3_get_status_name(initStatus_) << "\n";
+        return;
+      }
+      running_ = true;
+      if ( async_ ) {
+        throw std::runtime_error("Async not supported yet");
```

Comment:
Eventually this class would own an event loop thread that purely does IO. For now everything blocks in its current thread (S3lib claims to be fully threadsafe in this mode)